extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -713,11 +761,25 @@ export class JupyterServerInstallation implements IJupyterServerInstallation {
 		return path;
 	}
 
-	public static getPythonExePath(pythonInstallPath: string, useExistingInstall: boolean): string {
-		return path.join(
+	public static getPythonExePath(pythonInstallPath: string): string {
+		// The bundle version (0.0.1) is removed from the path for ADS-Python 3.8.8+.
+		// Only ADS-Python 3.6.6 contains the bundle version in the path.
+		let oldPythonPath = path.join(
+			pythonInstallPath,
+			'0.0.1',
+			process.platform === constants.winPlatform ? 'python.exe' : 'bin/python3');
+		let newPythonPath = path.join(
 			pythonInstallPath,
-			useExistingInstall ? '' : constants.pythonBundleVersion,
 			process.platform === constants.winPlatform ? 'python.exe' : 'bin/python3');
+
+		// Note: If Python exists in both paths (which can happen if the user chose not to remove Python 3.6 when upgrading),
+		// then we want to default to using the newer Python version.
+		if (!fs.existsSync(newPythonPath) && !fs.existsSync(oldPythonPath) || fs.existsSync(newPythonPath)) {
```

Comment:
Don't use sync fs functions in product code if you can help it - use the promise versions

Reply:
I noticed quite a few places in the jupyter installation code that uses fs.existsSync(). @corivera  do you have context around why we used sync? 
https://github.com/microsoft/azuredatastudio/blob/ed1db07f8cd4671587e534ba0d76c2081e154755/extensions/notebook/src/jupyter/jupyterServerInstallation.ts#L730
https://github.com/microsoft/azuredatastudio/blob/ed1db07f8cd4671587e534ba0d76c2081e154755/extensions/notebook/src/jupyter/jupyterServerInstallation.ts#L757

Reply:
Probably just old code, we weren't great about trying to enforce that originally.